### PR TITLE
Enable TimeProvider on New Model

### PR DESCRIPTION
**Pull request by FuPingFranco**
# Enable TimeProvider on New Model

## Description

Enable TimeProvider on new Wilson model.

### CODE REVIEW

**Comment by iNinja** [comment]
This looks good so far. If you could add the [FakeTimeProvider](https://learn.microsoft.com/en-us/dotnet/api/microsoft.extensions.time.testing.faketimeprovider?view=net-8.0) to the tests to prevent any flakiness, it would be perfect.

**Comment by FuPingFranco** [comment]
> This looks good so far. If you could add the [FakeTimeProvider](https://learn.microsoft.com/en-us/dotnet/api/microsoft.extensions.time.testing.faketimeprovider?view=net-8.0) to the tests to prevent any flakiness, it would be perfect.

I updated our new validation unit tests that required `DateTime.UtcNow` to use `TimeProvider` but decided to implement a `MockTimeProvider` instead as the `FakeTimeProvider` raises a new warning due to it being untested for net462.

**Comment by keegan-caruso** [comment]
> > This looks good so far. If you could add the [FakeTimeProvider](https://learn.microsoft.com/en-us/dotnet/api/microsoft.extensions.time.testing.faketimeprovider?view=net-8.0) to the tests to prevent any flakiness, it would be perfect.
> 
> I updated our new validation unit tests that required `DateTime.UtcNow` to use `TimeProvider` but decided to implement a `MockTimeProvider` instead as the `FakeTimeProvider` raises a new warning due to it being untested for net462.

I don't understand, https://www.nuget.org/packages/Microsoft.Extensions.TimeProvider.Testing/8.9.1#dependencies-body-tab

Has a net462 target.

File: build/dependenciesTest.props [hunk 1]
```diff
@@ -3,6 +3,7 @@
     <BannedApiAnalyzersVersion>3.3.4</BannedApiAnalyzersVersion>
     <DotNetCoreAppRuntimeVersion>2.1.30</DotNetCoreAppRuntimeVersion>
     <MicrosoftAzureKeyVaultCryptographyVersion>2.0.5</MicrosoftAzureKeyVaultCryptographyVersion>
+    <MicrosoftExtensionsTimeProviderTestingVersion>8.8.0</MicrosoftExtensionsTimeProviderTestingVersion>
```
Comment (keegan-caruso) [inline build/dependenciesTest.props]: Can we use latest?

Comment (keegan-caruso) [reply build/dependenciesTest.props]: Also this is unused, pending the other discussion about using the testing package.

Comment (iNinja) [reply build/dependenciesTest.props]: We should consider removing this until we need it.

**Review by iNinja — APPROVED** [review]
Looks good overall. Minor comment.

File: build/dependenciesTest.props (hunk 1 above)
Comment (FuPingFranco) [reply build/dependenciesTest.props]: Removed, I will follow-up with .NET team to understand the issue with this odd warning. Thanks Ignacio & Keegan!

**Review by keegan-caruso — APPROVED** [review]

**Pull request merged**